requester/print.go

```diff
@@ -99,54 +107,68 @@ func (r *report) finalize() {
 }
 
 func (r *report) print() {
-	sort.Float64s(r.lats)
-
 	if r.output == "csv" {
 		r.printCSV()
 		return
 	}
-
+	sort.Float64s(r.lats)
 	if len(r.lats) > 0 {
-		r.fastest = r.lats[0]
-		r.slowest = r.lats[len(r.lats)-1]
+		fastest, slowest := r.lats[0], r.lats[len(r.lats)-1]
 		fmt.Printf("Summary:\n")
 		fmt.Printf("  Total:\t%4.4f secs\n", r.total.Seconds())
-		fmt.Printf("  Slowest:\t%4.4f secs\n", r.slowest)
-		fmt.Printf("  Fastest:\t%4.4f secs\n", r.fastest)
+		fmt.Printf("  Slowest:\t%4.4f secs\n", slowest)
+		fmt.Printf("  Fastest:\t%4.4f secs\n", fastest)
 		fmt.Printf("  Average:\t%4.4f secs\n", r.average)
 		fmt.Printf("  Requests/sec:\t%4.4f\n", r.rps)
 		if r.sizeTotal > 0 {
 			fmt.Printf("  Total data:\t%d bytes\n", r.sizeTotal)
 			fmt.Printf("  Size/request:\t%d bytes\n", r.sizeTotal/int64(len(r.lats)))
 		}
-
-		if r.trace {
-			fmt.Printf("\nHttpTrace:\n")
-			fmt.Printf("  DNS+dialup:\t\t%4.4f secs\n", r.avgConn)
-			if r.avgDns > 0 {
-				fmt.Printf("  DNS lookup:\t\t%4.4f secs\n", r.avgDns)
-			}
-			fmt.Printf("  request Write:\t%4.4f secs\n", r.avgReq)
-			fmt.Printf("  response wait:\t%4.4f secs\n", r.avgDelay)
-			fmt.Printf("  response read:\t%4.4f secs\n", r.avgRes)
-		}
-
 		r.printStatusCodes()
-		r.printHistogram()
+		printHistogram(r.lats, "Response time", slowest, fastest)
 		r.printLatencies()
-	}
+		if len(r.errorDist) > 0 {
+			r.printErrors()
+		}
 
-	if len(r.errorDist) > 0 {
-		r.printErrors()
+		if r.trace {
+			fmt.Printf("\nDetailed Report:\n")
```

Comment:
We can avoid displaying histogram, instead display the average, fastest and slowest?